app/questionnaire/router.py

```diff
@@ -528,25 +521,23 @@ def _get_enabled_section_keys(
                 section_id
             ):
                 for list_item_id in self._list_store[repeating_list]:
-                    section_key = SectionKey(section_id, list_item_id)
-                    yield section_key
+                    yield SectionKey(section_id, list_item_id)
             else:
-                section_key = SectionKey(section_id, None)
-                yield section_key
+                yield SectionKey(section_id, None)
 
     def _get_first_incomplete_section_key(self) -> tuple[str, str | None] | None:
         for section_id, list_item_id in self._get_enabled_section_keys():
-            if not self._progress_store.is_section_or_repeating_blocks_progress_complete(
-                section_id, list_item_id
+            if not self._progress_store.is_section_complete(
+                SectionKey(section_id=section_id, list_item_id=list_item_id)
             ):
-                return section_id, list_item_id
+                return SectionKey(section_id=section_id, list_item_id=list_item_id)
```

Comment:
This is unpacking the key just to create it again.
This can be simplified to:
```py
        for section_key in self._get_enabled_section_keys():
            if not self._progress_store.is_section_complete(section_key):
                return section_key
```